specifications/api/security.md

```diff
@@ -0,0 +1,176 @@
+```
+[cardano-wallet.wallet-engine:Info:48] [2020-01-14 09:48:39.08 UTC] 135bfb99: Applying blocks [1139663.1 ... 1139663.1]
+[cardano-wallet.wallet-engine:Info:48] [2020-01-14 09:48:39.08 UTC] 135bfb99: Creating checkpoint at a212a0d3-[1139663.1#1]
+[cardano-wallet.wallet-engine:Info:48] [2020-01-14 09:48:39.09 UTC] 135bfb99: new wallet",''); DROP TABLE "wallet"; --, created at 2020-01-14 09:48:36.84066258 UTC, not delegating
+[cardano-wallet.wallet-engine:Info:48] [2020-01-14 09:48:39.09 UTC] 135bfb99: syncProgress: restored
+[cardano-wallet.wallet-engine:Info:48] [2020-01-14 09:48:39.09 UTC] 135bfb99: discovered 0 new transaction(s)
+[cardano-wallet.wallet-engine:Info:48] [2020-01-14 09:48:39.09 UTC] 135bfb99: local tip: a212a0d3-[1139663.1#1]
+[cardano-wallet.wallet-engine:Info:48] [2020-01-14 09:48:39.09 UTC] 135bfb99: blocks: [a212a0d3-[1139663.1#1] ∅]
+[cardano-wallet.wallet-engine:Info:48] [2020-01-14 09:48:39.09 UTC] 135bfb99: transactions: []
+```
+Hence, the name, creation date, delegation info of the wallet created earlier.
+
+Logging in lib/jormungandr/exe/cardano-wallet-jormungandr.hs, lib/core/src/Cardano/Wallet/Network.hs and lib/cli/src/Cardano/CLI.hs does not contribute to revealing of API calls info at all.
+
+**Conclusions** : The details of API logging is presented above. Sensitive information is concealed. Direct logging from API (as default it is Info severity level) is not giving very basic information about call method, endpoint, the response code, and time needed to realize it. Probably, it would be good to have Debug as a default - if we want to help other peoples and have maximum information possible without compromising their confidential info.
+
+2. Check what happens when the instance is long lived when it comes to logs.
+
+Logging is emitted to stdout and there is no way to change it now being the user of cardano-wallet. Daedalus, using cardano-wallet inside, realizes rotation based logging itself. It is set, user cannot control it. Moreover, it includes also logs from its layer. Some users, though, use cardano-wallet without Daedalus.
+
+**Conclusions** : Logs land to stdout, which has its pros and cons (it can be lost when terminal is set and by definition is less persistant). It would be good to add option for rotation, and embrace possibilities of iohk-monitoring-framework : https://github.com/input-output-hk/iohk-monitoring-framework/blob/bea0e079fc32ed316ce352d17d14199a680e3f6c/iohk-monitoring/src/Cardano/BM/Data/Rotation.lhs#L29
+Maybe it is even good to set this as default - users with problems could then immediately share logs with us.
+
+### OWASP risk : Security Misconfiguration
+
+1. Misconfigured HTTP headers
+Even is API is standalone piece of software, we need to think how it will behave being a part of HTML application. And here, XSS attacks against web browser clients can be realized. In order to avoid it a number of measures can be taken :
```

Comment:
```suggestion
Even if the API is a standalone piece of software, we need to think how it will behave as part of an HTML application. And here, XSS attacks against web browser clients can be realized. In order to avoid it a number of measures can be taken:
```

Reply:
corrected